xdsl/transforms/riscv_register_allocation.py

```diff
@@ -99,10 +286,11 @@ class RISCVRegisterAllocation(ModulePass):
 
     allocation_strategy: str = "GlobalJRegs"
 
-    def apply(self, ctx: MLContext, op: ModuleOp) -> None:
+    def apply(self, ctx: MLContext, op: ModuleOp, *args: Any) -> None:
```

Comment:
I'm not sure this is a good idea, could ModulePass arguments be used instead?

Reply:
Some regalloc strategies might have different parameters, so with the current API, you can either all make them `ModulePass` or you dispatch unused `ModulePass` parameters to all strategies.

Reply:
I don't get why they can't be stored on an instanec of this class